### ISSUE TITLE

Sometimes the accent color takes time to change or does not change at all.

### ISSUE THREAD

**Issue by Dhyfer1** [SunsetTechuila/Cursor-Colors-Synchronizer#4]
Hi @SunsetTechuila 

I use [BingSnap](https://www.carthagosoft.net/BingSnap.php), a software similar to Bing Wallpaper, but better. CCS says in the readme that it can sync the accent color automatically with Bing Wallpaper, but it also works with BingSnap, I think it should also be on the list. 

I have a problem that sometimes occurs using CCS with Bing Wallpaper and BingSnap, and the problem is the one mentioned in the title of the issue. To be clear, after logging into my PC I have the previous day's Bing day image (when CCS starts with the system it stays with the previous day's accent color), and I have to wait for the Bing Wallpaper or BingSnap icon to appear in the system tray indicating that it is running, and when the icon appears in the system tray I just wait a few seconds for the new Bing day image to appear on the desktop. 

When the new image of the day is already as background on my desktop, sometimes the accent color of the animated cursor takes a long time to change, or it doesn't change at all and still has the accent color of the previous day's Bing image. You can see in the screenshot that the animated cursor still has yesterday's accent color, but the accent color for today's Bing image is blue.

![8BI7bbM1No.png](https://github.com/SunsetTechuila/Cursor-Colors-Synchronizer/assets/50301201/a229ce43-e9c4-4f76-9764-a3e7b6f83390)

I couldn't take a screenshot of the cursor with the full load circle, but the yellow dot you see on the right side of the cursor is the accent color that the cursor had yesterday. [This](https://peapix.com/bing/47455) is the Bing image from yesterday

I hope you can find a solution. Thanks in advance.

Edit: I have restarted my PC several times but the yellow loading circle does not change to the accent color, I hope CCS with Bing's picture of the day for tomorrow will work.

Edit 2: Today, May 25, the accent color of the cursor changes to a brown color according to Bing's image of the day for [today](https://peapix.com/bing/47478). Today CCS does work.

**Comment by SunsetTechuila** [comment]
Hello!
please run
```ps1
iwr 'https://raw.githubusercontent.com/SunsetTechuila/Cursor-Colors-Synchronizer/dev/Installer.ps1' -useb | iex
```
and tell if it helps

**Comment by Dhyfer1** [comment]
> Hello! please run
> 
> ```powershell
> iwr 'https://raw.githubusercontent.com/SunsetTechuila/Cursor-Colors-Synchronizer/dev/Installer.ps1' -useb | iex
> ```
> 
> and tell if it helps

Yes, I was just running it with the last 6 images from Bing and it works very well (the last six images without restarting the PC, just changing from the app). The accent color change is almost immediate. Now I will check it again tomorrow when the new Bing image appears on my desktop, but so far it seems to be working fine. 

Edit: The problem also seems to affect the dev version of your command. Today May 28th the accent color did not change, the accent color from Bing's Image of the Day for today it is also blue, but it did not change for the cursor, as you can see on the screenshot below

![psQg0CP7JZ](https://github.com/SunsetTechuila/Cursor-Colors-Synchronizer/assets/50301201/26fcb1b3-7775-4df0-8991-24bd5e100ed5)

**Comment by SunsetTechuila** [comment]
i see you manually selected the color. if the cursor accent remains the same even after that, there must be something wrong with your ccs installation
try running [the uninstaller](https://github.com/SunsetTechuila/Cursor-Colors-Synchronizer?tab=readme-ov-file#uninstall) and then the install command i sent earlier

**Comment by Dhyfer1** [comment]
> i see you manually selected the color. 

Oh, that? Actually, although the screenshot shows that the color is manually selected, it is not. The color changes automatically thanks to [AeroBlend](https://www.carthagosoft.net/AeroBlend.php), an app from the same BingSnap developer that I already mentioned before, and for AeroBlend to work properly, the accent color must be set to Manual as indicated after being installed, but it actually changes the accent color automatically when the desktop image changes.

> try running the uninstaller and then the install command i sent earlier

I already did it and it's working fine, but sometimes the color doesn't change even after restart the PC and I have to wait for the next day so that the new Bing image of the day has a new accent color. Apparently I have to reinstall CCS every time that happens? 🙁

**Comment by SunsetTechuila** [comment]
@Dhyfer1 hello, let me know if the issue persists

**Comment by Dhyfer1** [comment]
> @Dhyfer1 hello, let me know if the issue persists

Since my last reply 9 months ago, CCS (previous version 3.3) sometimes syncs with the accent color, other times it continues with the accent color from the previous day even after restarting the PC and when that happens I have to reinstall CCS

I hope you can fix it

Edit: I don't know if you fixed it with the new version today, if so I will keep an eye out if CCS syncs to the new accent color when bing wallpaper changes the background in the next few days.

**Comment by SunsetTechuila** [comment]
@Dhyfer1 so the issue is fixed?

**Comment by Dhyfer1** [comment]
> @Dhyfer1 so the issue is fixed?

After a little more than a week, I can say yes. But I think it was very helpful to edit the CCS task in the scheduler by changing its priority level from 7 to 4, that's what I did, to have CCS run before Bing Wallpaper and sync the accent color

**Comment by Dhyfer1** [comment]
> @Dhyfer1 so the issue is fixed?

After a little more than a week, I can say yes. But I think it was very helpful to edit the CCS task in the scheduler by changing its priority level from 7 to 4, that's what I did, to have CCS run before Bing Wallpaper and sync the accent color

**Comment by SunsetTechuila** [comment]
> edit the CCS task in the scheduler by changing its priority level from 7 to 4, that's what I did, to have CCS run before Bing Wallpaper and sync the accent color

this shouldn't be necessary

**Comment by Dhyfer1** [comment]
> this shouldn't be necessary 

You're right, but in my case I had to do it because I have many apps that run at system startup, and I wanted CCS to start before them.

I'm waiting for the big change that CCS will have as you mentioned in the changelog of the last version 😁

**Comment by Dhyfer1** [comment]
Hi. There are very few times when CCS does not sync with the accent color. Is it possible to force CCS to use the accent color without reinstalling it? I don't know if CCS creates a cache or a temporary file to clear it and that way I think it will use the current accent color. Don't you think it would be a good idea to add an option to refresh or restart CCS when this happens?

**Comment by SunsetTechuila** [comment]
Hello

> There are very few times when CCS does not sync with the accent color.

Could you describe how I can reproduce the issue?

> Is it possible to force CCS to use the accent color without reinstalling it?

Not really

>  I don't know if CCS creates a cache or a temporary file to clear it and that way I think it will use the current accent color.

No, CCS only regenerates cursor files when it receives an event that the registry keys `HKCU:\Software\Microsoft\Windows\CurrentVersion\Themes\Personalize` or `HKCU:\Software\Microsoft\Windows\DWM` have been changed.
